tests/collapse-dtrace.rs

```diff
@@ -0,0 +1,183 @@
+    );
+    let expected_len = fs::metadata(expected_file)
+        .expect(&format!("Result file {} not found.", expected_file))
+        .len() as usize;
+    let mut result = Cursor::new(Vec::with_capacity(expected_len));
+    let mut perf = Folder::from(options);
+    let return_value = perf.collapse(r, &mut result);
+    let mut expected = BufReader::new(
+        File::open(expected_file).expect(&format!("Result file {} not found.", expected_file)),
+    );
+
+    result.set_position(0);
+
+    let mut buf = String::new();
+    let mut line_num = 1;
+    for line in result.lines() {
+        // Strip out " and ' since perl version does.
+        let line = line.unwrap().replace("\"", "").replace("'", "");
+
+        if expected.read_line(&mut buf).unwrap() == 0 {
+            println!("extra line: {}", line);
+            panic!(
+                "\noutput has more lines than expected result file: {}",
+                expected_file
+            );
+        }
+        assert_eq!(line, buf.trim_end(), "\n{}:{}", expected_file, line_num);
+        buf.clear();
+        line_num += 1;
+    }
+
+    if expected.read_line(&mut buf).unwrap() > 0 {
+        panic!(
+            "\n{} has more lines than output, beginning at line: {}",
+            expected_file, line_num
+        )
+    }
+
+    return_value
+}
+
+fn test_collapse_perf_logs<F>(input_file: &str, asserter: F)
+where
+    F: Fn(&Vec<testing_logger::CapturedLog>),
+{
+    test_collapse_perf_logs_with_options(input_file, asserter, Options::default());
+}
+
+fn test_collapse_perf_logs_with_options<F>(input_file: &str, asserter: F, options: Options)
+where
+    F: Fn(&Vec<testing_logger::CapturedLog>),
+{
+    testing_logger::setup();
+    let r = BufReader::new(File::open(input_file).unwrap());
+    let mut result = Cursor::new(vec![]);
+    let mut perf = Folder::from(options);
+    perf.collapse(r, &mut result).unwrap();
+    testing_logger::validate(asserter);
+}
+
+fn options_from_vec(opt_vec: Vec<&str>) -> Options {
+    let mut options = Options::default();
+    for option in opt_vec {
+        match option {
+            "addrs" => options.include_addrs = true,
+            "kernel" => options.annotate_kernel = true,
+            "all" => {
+                options.include_addrs = true;
+                options.annotate_kernel = true;
+            }
+            opt => panic!("invalid option: {}", opt),
+        }
+    }
+    options
+}
+
+// Create tests for test files in $dir. The test files are used as input
+// and the results are compared to result files in the results sub directory.
+// The test and result file names are derived from $name.
+// The part after the double underscore are underscore separated options.
+// For example, perf_cycles_instructions_01_pid will use the following:
+//     test file: perf-cycles-instructions-01.txt
+//     result file: perf-cycles-instructions-01-collapsed-pid.txt
+//     flag: pid
+macro_rules! collapse_perf_tests_inner {
```

Comment:
This can probably also be shared across all the collapse tests, right @jasonrhansen ?

Reply:
I removed this copy as it's no longer required.